roaring/roaring.go

```diff
@@ -3961,6 +3961,12 @@ func (op *op) UnmarshalBinary(data []byte) error {
 	_, _ = h.Write(data[0:9])
 
 	if op.typ > 1 {
+		// The maximum integer value for a int64 is 9223372036854775807
+		maxInt := 9223372036854775807
+		maxOpValue := maxInt/8-13
+		if maxOpValue < int(op.value) {
```

Comment:
This is an improvement, but there are a couple more things we can do:

- I'd prefer `maxInt64` over `maxInt`, but moreso, I would suggest directly using `math.MaxInt64`, (see https://golang.org/pkg/math/). Then we just have one line `maxOpValue := math.MaxInt64/8-13`, and the preceding comment is not really needed.
- We can move the definition of this outside the function, especially since there is already a similar value defined above (`minOpSize`), it can go right next to that.  

This is no longer relevant here, but just good to know: the value in the previous commit, `1152921504606847000` is not quite right. This looks like the rounded value you might get if you compute `MaxInt/8-13` in double precision floating point, rather than integer arithmetic. When working with 64-bit integers, this will often be a problem - be careful to use integer arithmetic operations in a situation like this, where you want both the input and the output to be an integer.

Reply:
The op.value isn't set until like 3958. Therefore the check has to go after that, right?

Reply:
That's right. I might have made an unclear suggestion: the definition of the bound value `maxOpValue := math.MaxInt64/8-13` can be moved, while keeping the check in the same place. `maxBatchSize` is a slightly more descriptive name.